src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,124 @@
+package duke.main;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+
+import duke.exception.DeadlineIncompleteException;
+import duke.exception.DeleteIncompleteException;
+import duke.exception.DeleteOutOfListException;
+import duke.exception.DoneIncompleteException;
+import duke.exception.DoneOutOfListException;
+import duke.exception.DukeException;
+import duke.exception.EventIncompleteException;
+import duke.exception.FindIncompleteException;
+import duke.exception.NoInputException;
+import duke.exception.TodoIncompleteException;
+import duke.exception.UnknownInputException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Parser is used to make sense the input from the user and convert it to executable command.
+ */
+public class Parser {
+
+    public static String[] commands = {"list", "done", "deadline", "event", "todo", "delete", "bye", "find"};
+    public static ArrayList<String> VALID_COMMAND = new ArrayList<>(Arrays.asList(commands));
```

Comment:
I think names of Collections should be plural?

Reply:
Yes, thanks for pointing that out!